File: 29. Divide Two Integers/divide_two_integer.java
public class divide_two_integer{
	public static class Quotient{
		public int integer_quotient;
		public int fraction;
		public int out_of;
		
		public void print (){
			System.out.print(integer_quotient);
			System.out.print("\t");
			System.out.print(fraction);
			System.out.print("\t");
			System.out.println(out_of);
		}
	}
	
	public static int division_whole (int number, int divisor){
		int out = 0;
		int running_number = number;
		
		while ((running_number - divisor) >= 0){
			out++;
			running_number -= divisor;
		}
		
		return (out);
	}
	
	public static int modulo_whole (int number, int divisor){
		int running_number = number;
		
		while ((running_number - divisor) >= 0){
			running_number -= divisor;
		}
		
		return (running_number);
	}
	
	public static int multiplication (int multiplicand, int multiplier){
		int out = 0;
		
		for (int i = 0; i < multiplier; i++){
			out += multiplicand;
		}
		
		return (out);
	}
	
	public static Quotient division (int number, int divisor){
		int no_of_figures = 10;
		
		int base = 10;
		int counter = 0;
		int fraction = 0;
		int integer_quotient = division_whole (number, divisor);
		int remainder = modulo_whole (number, divisor);
		int running_number = remainder;
		int out_of = 1;
		
		while ((running_number != 0) && (counter < no_of_figures)){
			//running_number *= base;
			running_number = multiplication(running_number, base);
			//out_of *= base;
			out_of = multiplication(out_of, base);
			
			if (fraction == 0){
				fraction += division_whole(running_number, divisor);
			} else {
				//fraction *= base;
				fraction = multiplication(fraction, base);
				fraction += division_whole(running_number, divisor);
			}
			
			running_number = modulo_whole(running_number, divisor);
			counter++;
		}
		
		Quotient o = new Quotient();
		o.integer_quotient = integer_quotient;
		o.fraction = fraction;
		o.out_of = out_of;
		
		return (o);
	}
	
	public static void main (String [] args){
		int number = 6402;
		int divisor = 8;
		
		Quotient q = division (number, divisor);
		
		q.print();
	}
}

File: 29. Divide Two Integers/divide_two_integer.swift
struct Quotient {
	var integer_quotient: Int
	var fraction: Int
	var out_of: Int
	
	func print_values () -> Void {
		print("This is the version from Swift");
		print(integer_quotient);
		print(fraction);
		print(out_of);
	}
}

func division_whole (number: Int, divisor: Int) -> Int {
	var out: Int = 0;
	var running_number: Int = number;
	
	while (running_number - divisor) >= 0{
		out += 1;
		running_number -= divisor;
	}
	
	return out;
}

func modulo_whole (number: Int, divisor: Int) -> Int {
	var running_number: Int = number;
	
	while (running_number - divisor) >= 0{
		running_number -= divisor;
	}
	
	return running_number;
}

func multiplication (multiplicand: Int, multiplier: Int) -> Int{
	var out = 0;
	
	for i in 0..<multiplier{
		out += multiplicand;
	}
	
	return out;
}

func division (number: Int, divisor: Int) -> Quotient{
	let no_of_figures: Int = 10;
	let base: Int = 10;
	var counter: Int = 0;
	var fraction: Int = 0;
	let integer_quotient = division_whole(number: number,divisor: divisor);
	let remainder = modulo_whole(number: number,divisor: divisor);
	var running_number = remainder;
	var out_of = 1;
	
	while ((running_number != 0) && (counter < no_of_figures)){
		//running_number *= base;
		running_number = multiplication(multiplicand: running_number, multiplier: base);
		//out_of *= base;
		out_of = multiplication(multiplicand: out_of, multiplier: base);
		
		if (fraction == 0){
			fraction += division_whole(number: running_number, divisor: divisor);
		} else {
			//fractioin *= base;
			fraction = multiplication(multiplicand: fraction, multiplier: base);
			
			fraction += division_whole(number: running_number,divisor: divisor);
		}
		
		running_number = modulo_whole(number: running_number,divisor: divisor);
		counter += 1;
	}
	
	return Quotient(integer_quotient: integer_quotient, fraction: fraction, out_of: out_of);
}

let number: Int = 6402;
let divisor: Int = 8;

let q: Quotient = division(number: number, divisor: divisor);

q.print_values();
